generators/client/templates/angular/src/main/webapp/app/_app.module.ts

```diff
@@ -73,9 +90,66 @@ import {
     ],
     providers: [
         ProfileService,
-        customHttpProvider(),
         PaginationConfig,
-        UserRouteAccessService
+        UserRouteAccessService,
+        <%_ if (authenticationType === 'jwt') { _%>
+        {
+            provide: HTTP_INTERCEPTORS,
+            useFactory: (localStorageService, sessionStorageService) => new AuthInterceptor(localStorageService, sessionStorageService),
+            multi: true,
+            deps: [
+                LocalStorageService,
+                SessionStorageService
+            ]
+        },
+        <%_ } _%>
+        <%_ if (authenticationType === 'jwt' || authenticationType === 'uaa') { _%>
+        {
+            provide: HTTP_INTERCEPTORS,
+            useFactory: (injector) => new AuthExpiredInterceptor(injector),
```

Comment:
Have you checked, this part? (I am not sure)AOT  is handling this correctly in the new version...

Reply:
can't we use `useClass` here?